main.go

```diff
@@ -46,15 +46,21 @@ func main() {
 	fmt.Printf("\nExecuting Matrix...\n\n")
 	results := execute.Run(plan)
 
-	reporter, err := output.GetReporter(config.Report)
+	reporter, err := output.GetReporter(config.Reports)
 	if err != nil {
 		fmt.Println(err)
 		os.Exit(1)
 	}
-	summary := reporter.Stream(config, results)
-	output.Summarize(summary)
 
-	if summary.Failed == true {
+	if err = reporter.Start(config); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
nit - perhaps we should move these 2 lines to a `Fail` function than can be used in the few cases below.